src/rpc/rpc_reward_api.py

```diff
@@ -376,8 +380,13 @@ def do_rpc_request(self, request, time_out=120):
                 "RPC URL '{}' not found. Is this node in archive mode?".format(request)
             )
 
-        # URL returned something broken
-        if resp.status_code != HTTPStatus.OK:
+        # URL returned something broken from the client side 4xx
+        # server side errors 5xx can pass for a retry
+        if (
+            HTTPStatus.BAD_REQUEST
+            <= resp.status_code
```

Comment:
If the response status code is greater or equal to BAD_REQUEST (400) and less then INTERNAL_SERVER_ERROR (500) the raise the APIProvider Exception which won't trigger a retry in the while loop.